tests/FSharp.Compiler.ComponentTests/Conformance/BasicGrammarElements/MethodResolution.fs

```diff
@@ -77,7 +77,7 @@ extends [runtime]System.Object
 
   }
 
-  .method assembly specialname static class [runtime]System.Tuple`2<bool,int32> get_patternInput@9() cil managed
+  .method public specialname static class [runtime]System.Tuple`2<bool,int32> get_patternInput@9() cil managed
```

Comment:
@KevinRansom so these guys became public now... I was pondering on this - I don't really have enough picture to evaluate the real world implications since this is quite a special case, but what I _feel_ is that this is _either_ okay _or_ uncovering a different issue which should be addressed separately.

Reply:
@psfinaki public is the expected scope, it is not hard to argue that it is incorrect, but the bug is certainly elsewhere.

````
module OutOptionalTests

// unnecessary code elided here

let (_:bool), (_:int) = Thing.Do(i = 42)
let (_:bool), (_:int) = Thing.Do()
````
So:
1: bindings on modules default to being public ...
2: The binding is unnamed discards, so the side effect is still necessary.
if you make the bindings look like:
````
let (_a:bool), (_b:int) = Thing.Do(i = 42)
let (_c:bool), (_d:int) = Thing.Do()
````
Then compile with --allsigs you get this output:
````
module OutOptionalTests

// unnecessary code elided here

val _b: int
val _a: bool
val _d: int
val _c: bool
````
So give them a name and the value is emitted.  So ... if it is expected to not be public ... and I think it was relying on the side-effect of a bug.  Raise an issue suggesting that unnamed bindings should not be public when emitted to il.
And require that it's fixed before we merge this one.  I will look at it.  It is better as a separate issue, because they are entirely unrelated.

Reply:
@psfinaki - ignore the above over the weekend I revisited this and decided that my earlier approach was a bit too much of a hack.  I replaced it with a better more targetted fix.